SearchIT.testSearchVersionsByMetadata fails consistently with Elasticsearch storage

## Description

`SearchIT.testSearchVersionsByMetadata` fails consistently when running against
Elasticsearch storage. The test creates a version with metadata (name), then searches
for it by name, but the search returns 0 results. This is not a flaky/timing issue —
it fails across all three Elasticsearch test configurations.

## Environment

- **Registry version:** 3.3.1
- **Test group:** `search`
- **Storage:** Elasticsearch (search profile)

## Affected Configurations

All three Elasticsearch runs failed with the same error:

| Step | OCP | ES Version | Result |
|------|-----|------------|--------|
| Elasticsearch 9.0 (OCP Latest) | 4.22.2 | 9.0.0 | 6/7 passed, 1 failed |
| Elasticsearch 9.2 (OCP Latest) | 4.22.2 | 9.2.2 | 6/7 passed, 1 failed |
| Elasticsearch 9.2 (OCP 4.16) | 4.16.55 | 9.2.2 | 6/7 passed, 1 failed |

The other 6 tests in `SearchIT` all pass: `testSearchVersionsByLabels`,
`testSearchVersionsByContent`, `testSearchVersionsByDescription`,
`testSearchResultsFieldMapping`, etc.

## Error

```
[ERROR] io.apicurio.tests.search.SearchIT.testSearchVersionsByMetadata
        -- Time elapsed: 22.38 s <<< FAILURE!

org.opentest4j.AssertionFailedError:
  Failed handle callable: ...
  [org.opentest4j.AssertionFailedError:
    Expected at least 1 result when searching by name
    ==> expected: <true> but was: <false>]
  ==> expected: <true> but was: <false>
    at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:41)
    at io.apicurio.tests.ApicurioRegistryBaseIT.retry(ApicurioRegistryBaseIT.java:388)
    at io.apicurio.tests.ApicurioRegistryBaseIT.retry(ApicurioRegistryBaseIT.java:363)
    at io.apicurio.tests.ApicurioRegistryBaseIT.retry(ApicurioRegistryBaseIT.java:348)
    at io.apicurio.tests.ApicurioRegistryBaseIT.retry(ApicurioRegistryBaseIT.java:334)
    at io.apicurio.tests.search.SearchIT.testSearchVersionsByMetadata(SearchIT.java:119)
```

The `retry()` helper retried the search multiple times (test took 22.38s) but never
received any results, indicating this is not a simple indexing delay.

## Analysis

This failure is **deterministic**, not flaky — it reproduces across both ES versions
(9.0 and 9.2) and both OCP versions (4.16 and 4.22). The fact that the other search
tests pass (search by labels, content, description) narrows the issue to the
search-by-metadata/name code path specifically.

Possible causes:
1. The metadata/name field is not being indexed correctly in Elasticsearch
2. The search query for metadata name uses a different field mapping or query type
   that doesn't match what's stored
3. A regression in how version metadata (as opposed to labels or content) is
   persisted to the search index

## Non-Elasticsearch Profiles

This test is only run with the `search` test group (`-Dgroups=search`), which is
exclusive to the Elasticsearch deployment profile. The other profiles (inmemory,
postgresql, mysql, kafkasql) run different test groups and all passed.

--- a/integration-tests/src/test/java/io/apicurio/tests/search/SearchIT.java
+++ b/integration-tests/src/test/java/io/apicurio/tests/search/SearchIT.java
@@ -115,11 +115,13 @@ public class SearchIT extends ApicurioRegistryBaseIT {
                     results.getVersions().get(0).getArtifactId());
         });
 
-        // Search by name
+        // Search by name. Name matching is exact unless the value carries explicit '*'
+        // wildcards (see SqlSearchRepository.buildNameClause(), #8002), so a partial
+        // value needs a trailing wildcard to match "Search Smoke Test API".
         retry(() -> {
             VersionSearchResults results = registryClient.search().versions().get(config -> {
                 config.queryParameters.groupId = GROUP;
-                config.queryParameters.name = "Search Smoke Test";
+                config.queryParameters.name = "Search Smoke Test*";
             });
             Assertions.assertTrue(results.getCount() >= 1,
                     "Expected at least 1 result when searching by name");

